src/main/java/io/cosmosoftware/kite/instrumentation/Scenario.java

```diff
@@ -7,148 +7,186 @@
-              "runCommand(). Command:\r\n"
-                  + command
-                  + "\r\n"
-                  + ReportUtils.getStackTrace(e));
-          result.append("  Error ").append(e.getMessage());
-        }
-        result.append("\n\n");
+    result.append("Running ").append(this.command).append(" on ");
+    if (this.type.equals("gateway")) {
+      result.append(this.gateway);
+      logger.info(result.toString());
+      result.append(this.runCommandGateway(this.command, GridId, instrumentUrl, instrumentation));
+    }
+    if (this.type.equals("client")) {
+      String sessionId = ((RemoteWebDriver) webDriver).getSessionId().toString();
+      String nodeIp = TestUtils.getPrivateIp(remoteAddress, sessionId);
+      result.append("client ").append(nodeIp);
+      if (((RemoteWebDriver) webDriver).getCapabilities().getPlatform().toString().equalsIgnoreCase("LINUX")) {
+        result.append(this.runCommandClient(this.command, GridId, instrumentUrl, nodeIp));
+      } else {
+        result.append("Node ").append(nodeIp).append(" is not Linux");
       }
     }
     return result.toString();
   }
 
-  public String cleanUp() {
+  public String cleanUp(WebDriver webDriver, Instrumentation instrumentation, String remoteAddress, String GridId, String instrumentUrl) {
     StringBuilder result = new StringBuilder();
-    for (String gw : this.commandList.keySet()) {
-      String command = this.commandList.get(gw);
-      if (!command.equals("")) {
-        Instance instance = instrumentation.get(gw);
-        String[] interfacesList = {instance.getNit0(), instance.getNit1(), instance.getNit2()};
-        for (String inter : interfacesList) {
-          if (command.contains(inter)) {
-            String cleanUpCommand = "sudo tc qdisc del dev " + inter + " root";
-            try {
-              SSHManager sshManager = new SSHManager(instance.getKeyFilePath(), instance.getUsername(),
-                  instance.getIpAddress(), cleanUpCommand);
-              if (sshManager.call().commandSuccessful()) {
-                Thread.sleep(1000);
-                logger.info("cleanUp() : " + inter + " on gateway " + gw);
-                result.append(cleanUpCommand);
-              } else {
-                logger.error("Failed cleanUp() : " + inter + " on gateway " + gw);
-                result.append("  FAILURE (Client : ").append(instance.getIpAddress()).append(")");
-              }
-            } catch (Exception e) {
-              logger.error(
-                  "cleanUp(). \r\n"
-                      + inter
-                      + "\r\n"
-                      + ReportUtils.getStackTrace(e));
-              result.append("  Error ").append(e.getMessage());
-            }
-            result.append(" on gateway ").append(gw).append("\n\n");
-          } else {
-            logger.info("No CleanUp to do on interface : " + inter + " on gateway " + gw);
-          }
-        }
+    String inter = "enp0s8";
```

Comment:
why is this hardcoded here?

Reply:
I don't really know how to handle interface for now ...

I think if I have an Instrumentation I take it from there, if the KiteServer create the grid I hardcode it assuming it is always the same and I have also manual command to handle

Reply:
ok, then declare at the top of the class. we'll probably change that at some point

```java
private static final String INTERFACE_0_NAME = "enp0s8";
private static final String INTERFACE_1_NAME = "enp0s9";
```